streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -493,28 +549,45 @@ public void closeAndRecycleState() {
     private Map<TopicPartition, Long> prepareClose(final boolean clean) {
         final Map<TopicPartition, Long> checkpoint;
 
-        if (state() == State.CREATED) {
-            // the task is created and not initialized, just re-write the checkpoint file
-            checkpoint = Collections.emptyMap();
-        } else if (state() == State.RUNNING) {
-            closeTopology(clean);
+        switch (state()) {
```

Comment:
Out of the scope of this PR: why we need to return the checkpoint map to bookkeep at the `task-manager`? It seems we just re-distribute it in the `close` call.

I think we do not need to expose the checkpoint map in task-manager eventually?

Reply:
Not 100% sure what the end state will be atm, but yes, we should be able to simplify this further in follow up PRs.